tasks/use_draftlogs.js

```diff
@@ -0,0 +1,77 @@
+var fs = require('fs');
+var path = require('path');
+var constants = require('./util/constants');
+
+var pathToDraftlogs = constants.pathToDraftlogs;
+var pathToChangelog = constants.pathToChangelog;
+
+var chZero = '0'.charCodeAt(0);
+var chNine = '9'.charCodeAt(0);
+
+function startsWithNumber(v) {
+    var ch = v.charCodeAt(0);
+    return chZero <= ch && ch <= chNine;
+}
+
+var allLogs = fs.readdirSync(pathToDraftlogs).filter(startsWithNumber);
+
+var len = allLogs.length;
+if(!len) return;
+
+var writeAfterMe = 'where X.Y.Z is the semver of most recent plotly.js release.';
+var changelog = fs.readFileSync(pathToChangelog).toString().split(writeAfterMe);
+var head = changelog[0];
+var foot = changelog[1];
+
+var all = {
+    Added: [],
+    Removed: [],
+    Deprecated: [],
+    Changed: [],
+    Fixed: []
+};
+
+var ENTER = '\n';
+
+for(var i = 0; i < len; i++) {
+    var filename = allLogs[i];
+    var message = fs.readFileSync(path.join(pathToDraftlogs, filename), { encoding: 'utf-8' }).toString();
+    // trim empty lines
+    message = message.split(ENTER).filter(function(e) { return !!e; }).join(ENTER);
+
+    if(filename.endsWith('_add.md')) {
+        all.Added.push(message);
+    } else if(filename.endsWith('_remove.md')) {
+        all.Removed.push(message);
+    } else if(filename.endsWith('_deprecate.md')) {
+        all.Deprecated.push(message);
+    } else if(filename.endsWith('_change.md')) {
+        all.Changed.push(message);
+    } else if(filename.endsWith('_fix.md')) {
+        all.Fixed.push(message);
+    }
```

Comment:
Else throw an error, logging the filename as having been ignored.

Might want to also loosen the `endsWith` tests - to eg `filename.toLowerCase().indexOf('_add') !== -1` so stuff like `5678_Added.md` will still be picked up.

Reply:
I think we have a test which disallows use of uppercase in filenames.

Reply:
We [do test for uppercase](https://github.com/plotly/plotly.js/blob/7374eb8efd9ff9fa7b38ec24b0c1b5e9f953fd70/tasks/test_syntax.js#L210-L216), but I don't think it'll dig into the `draftlogs` dir. And there's still add vs adds vs added etc. Anyway the main thing is to throw an error if we ignore the file. After adding that if you still want to be strict about names that's ok, it'll just occasionally make more work during release if we get an incorrect name.

Reply:
Good call. Addressed in https://github.com/plotly/plotly.js/pull/5780/commits/3a91c0e380886d6292b12a71b22fd25d298b5f82.